instrumentation/apache-httpclient/apache-httpclient-2.0/javaagent/src/test/groovy/CommonsHttpClientLatestDepsTest.groovy

```diff
@@ -0,0 +1,125 @@
+import io.opentelemetry.instrumentation.test.base.HttpClientTest
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes
+import org.apache.commons.httpclient.HttpClient
+import org.apache.commons.httpclient.HttpConnectionManager
+import org.apache.commons.httpclient.HttpMethod
+import org.apache.commons.httpclient.MultiThreadedHttpConnectionManager
+import org.apache.commons.httpclient.methods.DeleteMethod
+import org.apache.commons.httpclient.methods.GetMethod
+import org.apache.commons.httpclient.methods.HeadMethod
+import org.apache.commons.httpclient.methods.OptionsMethod
+import org.apache.commons.httpclient.methods.PostMethod
+import org.apache.commons.httpclient.methods.PutMethod
+import org.apache.commons.httpclient.methods.TraceMethod
+import spock.lang.IgnoreIf
+import spock.lang.Shared
+
+//this test will be ignored if not executed with -PtestLatestDeps=true
+//because the latest dependency commons-httpclient v3.1 allows a call to the executeMethod 
+//with some null parameters like HttpClient.executeMethod(null, request, null)  
+//but this construct is not allowed in commons-httpclient v2 that is used for regular otel testing
+@IgnoreIf({ !Boolean.getBoolean("testLatestDeps") })
+class CommonsHttpClientLatestDepsTest extends HttpClientTest<HttpMethod> implements AgentTestTrait {
+  @Shared
+  HttpConnectionManager connectionManager = new MultiThreadedHttpConnectionManager()
+  @Shared
+  HttpClient client = buildClient(false)
+  @Shared
+  HttpClient clientWithReadTimeout = buildClient(true)
+
+  def buildClient(boolean readTimeout) {
+    HttpClient client = new HttpClient(connectionManager)
+    client.setConnectionTimeout(CONNECT_TIMEOUT_MS)
+    if (readTimeout) {
+      client.setTimeout(READ_TIMEOUT_MS)
+    }
+    return client
+  }
+
+  HttpClient getClient(URI uri) {
+    if (uri.toString().contains("/read-timeout")) {
+      return clientWithReadTimeout
+    }
+    return client
+  }
+
+  @Override
+  HttpMethod buildRequest(String method, URI uri, Map<String, String> headers) {
+    def request
+    switch (method) {
+      case "GET":
+        request = new GetMethod(uri.toString())
+        break
+      case "PUT":
+        request = new PutMethod(uri.toString())
+        break
+      case "POST":
+        request = new PostMethod(uri.toString())
+        break
+      case "HEAD":
+        request = new HeadMethod(uri.toString())
+        break
+      case "DELETE":
+        request = new DeleteMethod(uri.toString())
+        break
+      case "OPTIONS":
+        request = new OptionsMethod(uri.toString())
+        break
+      case "TRACE":
+        request = new TraceMethod(uri.toString())
+        break
+      default:
+        throw new IllegalStateException("Unsupported method: " + method)
+    }
+    headers.each { request.setRequestHeader(it.key, it.value) }
+    return request
+  }
+
+  @Override
+  int sendRequest(HttpMethod request, String method, URI uri, Map<String, String> headers) {
```

Comment:
WDYT about extracting a common base class for this class and `CommonsHttpClientTest`. I believe this is the only method that is different in them.

Reply:
of course this makes sense.

I hope the naming will be fine, otherwise just let me know.
Thank  you for reviewing this.